akka-stream/src/main/scala/akka/stream/Attributes.scala

```diff
@@ -293,6 +295,104 @@ object Attributes {
       extends Attribute
   final case object AsyncBoundary extends Attribute
 
+  /**
+   * Cancellation strategies provide a way to configure the behavior of a stage when `cancelStage` is called.
+   *
+   * It is only relevant for stream components that have more than one output and do not define a custom cancellation
+   * behavior by overriding `onDownstreamFinish`. In those cases, if the first output is cancelled, the default behavior
+   * is to call `cancelStage` which shuts down the stage completely. The given strategy will allow customization of how
+   * the shutdown procedure should be done precisely.
+   */
+  @ApiMayChange
+  final case class CancellationStrategy(strategy: CancellationStrategy.Strategy) extends MandatoryAttribute
+  @ApiMayChange
+  object CancellationStrategy {
+    private[stream] val Default: CancellationStrategy = CancellationStrategy(CompleteStage)
+    @ApiMayChange
+    object SiblingDownstreamWasCancelled extends RuntimeException("Sibling downstream was cancelled") with NoStackTrace
```

Comment:
this is not in master so I think it will cause mima check problem

Reply:
That's a good point. The easiest would be to make the concrete type private. That makes it somewhat hard to handle this situation but it is still an improvement over the previous version.

Alternatively, we could also already backport the exception from master, which would be `SubscriptionWithCancellationCause.NoMoreDataNeeded`.

Reply:
Did the second option.